src/backend/core/api/viewsets.py

```diff
@@ -576,8 +576,15 @@ def perform_create(self, serializer):
         """Add a new access to the document and send an email to the new added user."""
         access = serializer.save()
         language = self.request.headers.get("Content-Language", "en-us")
+        
+        username_sender = self.request.user.full_name
+        email_sender = self.request.user.email
+        if(not username_sender):
+            username_sender = self.request.user.email
+            email_sender = ""
+
         access.document.email_invitation(
-            language, access.user.email, access.role, self.request.user.email
+            language, access.user.email, access.role, username_sender, email_sender
```

Comment:
Semantically, shouldn't it be: 
```suggestion
            language, access.user.email, access.role, sender_name, sender_email
```

The term "username" is usually used with another meaning.